pydantic/dataclasses.py

```diff
@@ -128,18 +203,23 @@ def dataclass(
     unsafe_hash: bool = False,
     frozen: bool = False,
     config: Type[Any] = None,
-) -> Union[Callable[[Type[Any]], 'DataclassType'], 'DataclassType']:
+) -> Union[Callable[[Type[Any]], Type['DataclassType']], Type['DataclassType']]:
```

Comment:
Wait, surely this signature is now wrong, you have double `Type` and the type of a type is always just `type`.

Maybe `DataclassType` is defined wrongly?

Reply:
The name was missleading as `DataclassType` was refering to an instance of a pydantic dataclass. I renamed it to `Dataclass`.